uplink/storage/streams/store_test.go

```diff
@@ -1,395 +1,226 @@
+		require.NoError(t, err)
+
+		objects := []struct {
+			path    string
+			content []byte
+		}{
+			{"aaaa/afile1", []byte("content")},
+			{"aaaa/bfile2", []byte("content")},
+			{"bbbb/afile1", []byte("content")},
+			{"bbbb/bfile2", []byte("content")},
+			{"bbbb/bfolder/file1", []byte("content")},
 		}
-
-		gomock.InOrder(calls...)
-
-		streamStore, err := NewStreamStore(mockSegmentStore, segSize, newStore(), encBlockSize, dataCipher, inlineSize)
-		if err != nil {
-			t.Fatal(err)
+		for _, test := range objects {
+			test := test
+			data := bytes.NewReader(test.content)
+			path := storj.JoinPaths(bucketName, test.path)
+			_, err := streamStore.Put(ctx, path, storj.EncNull, data, []byte{}, expiration)
+			require.NoError(t, err)
 		}
 
-		ranger, meta, err := streamStore.Get(ctx, test.path, pathCipher)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		assert.Equal(t, test.streamRanger.Size(), ranger.Size(), errTag)
-		assert.Equal(t, test.streamMeta, meta, errTag)
-	}
+		prefix := bucketName
+
+		// should list all
+		items, more, err := streamStore.List(ctx, prefix, "", "", storj.EncNull, true, 10, meta.None)
+		require.NoError(t, err)
+		require.False(t, more)
+		require.Equal(t, len(objects), len(items))
+
+		// should list first two and more = true
+		items, more, err = streamStore.List(ctx, prefix, "", "", storj.EncNull, true, 2, meta.None)
+		require.NoError(t, err)
+		require.True(t, more)
+		require.Equal(t, 2, len(items))
+
+		// // should list only prefixes
+		items, more, err = streamStore.List(ctx, prefix, "", "", storj.EncNull, false, 10, meta.None)
+		require.NoError(t, err)
+		require.False(t, more)
+		require.Equal(t, 2, len(items))
+
+		// should list only BBBB bucket
+		prefix = storj.JoinPaths(bucketName, "bbbb")
+		items, more, err = streamStore.List(ctx, prefix, "", "", storj.EncNull, false, 10, meta.None)
+		require.NoError(t, err)
+		require.False(t, more)
+		require.Equal(t, 3, len(items))
+
+		// should list only BBBB bucket after afile
+		items, more, err = streamStore.List(ctx, prefix, "afile1", "", storj.EncNull, false, 10, meta.None)
+		require.NoError(t, err)
+		require.False(t, more)
+		require.Equal(t, 2, len(items))
+
+		// should list nothing
+		prefix = storj.JoinPaths(bucketName, "cccc")
+		items, more, err = streamStore.List(ctx, prefix, "", "", storj.EncNull, true, 10, meta.None)
+		require.NoError(t, err)
+		require.False(t, more)
+		require.Equal(t, 0, len(items))
+	})
 }
 
-func TestStreamStoreDelete(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
+func runTest(t *testing.T, test func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet, streamsStore streams.Store)) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		// TODO move apikey creation to testplanet
```

Comment:
out of curiosity, why this?

Reply:
copy/paste from refactored test. Its about making api key creation and access easier. To add apikey you need to make a lot of steps in test, see code below.